### ISSUE TITLE
When resizing a window using the mouse, the cursor jumps to the top left of the window

### ISSUE THREAD

**Issue by Hubro** [i3/i3#3598]
## I'm submitting a…

<pre>
[x] Bug
[ ] Feature Request
[ ] Documentation Request
[ ] Other (Please describe in detail)
</pre>

## Current Behavior

When I hold the Super key and use the right mouse button in order to resize a window, the mouse jumps to the top left of the screen. This happens on version 4.16 as well as a fresh build from the "next" branch, and it happens on two completely separate PCs, one running Manjaro, the other running Arch. Both systems are completely up-to-date and running the base auto-generated i3 config.

After Googling, the only other reported case of this issue I could find was this post from 2014, with a video describing the issue: https://faq.i3wm.org/question/4661/mouse-jumps-when-resizing-window/index.html

The only difference between his video and my issue is that the mouse doesn't necessarily jump to the absolute top left of the screen (0x0), but ends up between 0 and 100 px from the top left edge of the screen, based on where my mouse was before it jumped. Checking the debug logs (see below) it looks like the X and Y position of the cursor is divided by 10 for some reason.

## Expected Behavior

I would expect the mouse to not jump to the top left of the screen.

## Reproduction Instructions

Open two terminal windows and dock them horizontally. Place the mouse between the two terminal windows and press Super+Right click. As soon as the mouse button is pressed down, the cursor jumps to the edge between the windows. As soon as the mouse is *moved*, the cursor jumps to the top left of the screen.

## Environment

Output of `i3 --moreversion 2>&-`:
<pre>
Binary i3 version:  4.16 (2018-11-04) © 2009 Michael Stapelberg and contributors
Running i3 version: 4.16 (2018-11-04) (pid 1430)o abort…)
Loaded i3 config: /home/tomas/.config/i3/config (Last modified: Mon 07 Jan 2019 10:16:37 PM CET, 1234965 seconds ago)

The i3 binary you just called: /usr/bin/i3
The i3 binary you are running: i3
</pre>

The issue happens regardless of i3 config. It happens with my custom config file and with the automatically generated one.

<pre>
Logfile URL: https://logs.i3wm.org/logs/5674610515247104.bz2
</pre>

I believe these lines in the log file demonstrate the problem pretty well:

<pre>
01/22/2019 05:22:06 AM - resize.c:resize_callback:29 - new x = 1592, y = 484
01/22/2019 05:22:06 AM - resize.c:resize_callback:29 - new x = 158, y = 47
</pre>

Notice that as soon as I move the mouse, the mouse position moves to approximately 1/10th of the X and Y values.

<pre>
- Linux Distribution & Version: Manjaro 18.0.2 and Arch Linux (up to date)
- Are you using a compositor (e.g., xcompmgr or compton): no
</pre>

**Comment by Hubro** [comment]
Ok, right away after reading the debug log the 1/10th mouse position rang a bell. I just remembered why. I set my mouse sensitivity to 1/10th using the "Coordinate Transformation Matrix" property:

    $ xinput --set-prop "Logitech USB Receiver Mouse" "Coordinate Transformation Matrix" 0.100000, 0.000000, 0.000000, 0.000000, 0.100000, 0.000000, 0.000000, 0.000000, 1.000000

This is the only way I've found to reduce the mouse sensitivity in Linux, and it's essential to the usability of my system.

Apparently this transformation matrix also causes i3 to move the mouse to 1/10th of the X and Y coordinates when starting a window resize using the mouse.

**Comment by orestisf1993** [comment]
Code responsible for this:
https://github.com/i3/i3/blob/a9e476e899f15d1d874ec2e5ee8ebda6cf4b0c77/src/resize.c#L182-L183
https://github.com/i3/i3/blob/a9e476e899f15d1d874ec2e5ee8ebda6cf4b0c77/src/resize.c#L188-L189

are we supposed to apply the coordinate transformation matrix here?

**Comment by Airblader** [comment]
We could try it out :-)

**Comment by orestisf1993** [comment]
If we continue with https://github.com/i3/i3/pull/3559#discussion_r270268426 this will be resolved too

**Comment by orestisf1993** [comment]
I am not sure if this is our fault though. In whichever window manager, set the coordinate transformation matrix and run `xdotool mousemove 800 300`. The cursor is wrapped correctly (like in i3) but after any mouse movement it jumps to 80 30.

**Comment by Airblader** [comment]
My instinct also is that i3 shouldn't have to handle this at all. I would expect the server to do this transparently for clients. I'm not sure about it, though.

**Comment by orestisf1993** [comment]
I initially thought that it didn't apply here but I think we are hitting this bug: https://gitlab.freedesktop.org/xorg/xserver/issues/600

**Comment by gardockt** [comment]
It looks like the correct way to set a mouse sensitivity is by enabling flat acceleration.

From [libinput documentation about pointer acceleration](https://wayland.freedesktop.org/libinput/doc/latest/pointer-acceleration.html):
> libinput currently supports two profiles: “adaptive” and “flat”. The adaptive profile is the default profile for all devices and takes the current speed of the device into account when deciding on acceleration. The flat profile is simply a constant factor applied to all device deltas, regardless of the speed of motion.

You can do that by setting `Accel Profile Enabled` to `0, 1` (or `AccelProfile` to `"flat"` when editing `xorg.conf.d`) and `Accel Speed` to a number between -1 and 1, where 0 is the default sensitivity.

**Comment by Hubro** [comment]
@gardockt Yeah, that's what I've done this past year. This is the contents of my `/etc/X11/xorg.conf.d/99-mouse.conf`:

```
Section "InputClass"
	Identifier "Logitech G900 Wireless"
	MatchDriver "libinput"
	MatchProduct "Logitech G900"
	MatchIsPointer "on"
	Option "AccelProfile" "flat"
	Option "AccelSpeed" "-0.8"
EndSection

Section "InputClass"
	Identifier "Logitech G900 Wired"
	MatchDriver "libinput"
	MatchProduct "Logitech Gaming Mouse G900"
	MatchIsPointer "on"
	Option "AccelProfile" "flat"
	Option "AccelSpeed" "-0.8"
EndSection
```

**Comment by fghsgh** [comment]
> > libinput currently supports two profiles: “adaptive” and “flat”. The adaptive profile is the default profile for all devices and takes the current speed of the device into account when deciding on acceleration. The flat profile is simply a constant factor applied to all device deltas, regardless of the speed of motion.
> 
> You can do that by setting `Accel Profile Enabled` to `0, 1` (or `AccelProfile` to `"flat"` when editing `xorg.conf.d`) and `Accel Speed` to a number between -1 and 1, where 0 is the default sensitivity.

This does not work for touchpads handled by libinput, because libinput doesn't have the `Accel Profile Enabled` option nor the `Accel Speed` option for touchpads. It seems this was an intentional decision, although it is unclear to me why this would be the case. For touchpads handled by Synaptics, these also lack any similar options.
